x/slashing/src/types/signing_info.rs

```diff
@@ -0,0 +1,30 @@
+use gears::{tendermint::types::time::Timestamp, types::address::ConsAddress};
+use serde::{Deserialize, Serialize};
+
+/// ValidatorSigningInfo defines a validator's signing info for monitoring their
+/// liveness activity.
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+pub struct ValidatorSigningInfo {
+    pub address: ConsAddress,
+    /// Height at which validator was first a candidate OR was unjailed
+    pub start_height: i64,
```

Comment:
Same as above, could it be a `u32`? Same for `index_offset` and `missed_block_counter`.

Reply:
We can change these fields but I suggest that is better to keep original `i64` type. 

I suggest it because these structures are stored into a store during genesis.
For example, it is a call of method [`set_validator_signing_info`](https://github.com/rumos-io/gears/pull/134/files#diff-23123a007b59eddb57ccb8b58afcf300de915aabae27b715e39399f1e0aa6672R72). It stores full struct `ValidatorSigningInfo`.

Reply:
I think that we have "silent agreement" to resolve such issues with [dto's](https://en.wikipedia.org/wiki/Data_transfer_object)